### PR TITLE
added minimal pytest with example rst file

### PR DESCRIPTION
**Pull request by shimwell**
Thanks for making this super useful package

I have checked the other PR #20 using this minimal pytest code.

Perhaps this could be a start for testing

Here you can see the tests fail on the current master branch and pass on the branch with the fix for sphinx v4

Screenshot from 2021-05-10 16-06-48

### CODE REVIEW

**Comment by Peque** [comment]
Good initiative! :blush: 

A couple of comments:

- Can you put everything under `tests/`? In example, you could create a `tests/example/conf.py` file.
- Since you use `pytest` (I prefer that too :blush:), can you remove the `unittest` import and the `class` definition in the test?

Also, it is nice to have tests to run locally, but it is even better to have them running in CI! (i.e.: with Travis or GitHub Actions). I have another repo that you may check if you feel brave enough to go that extra mile. :stuck_out_tongue: 

https://github.com/markdownreveal/markdownreveal

(check out the `tox.ini` file and `.travis.yml` file if you feel interested)

**Comment by shimwell** [comment]
Yep I can give that a go.

Might even get a git squash out of me at the end :stuck_out_tongue_winking_eye: 

I believe GitHub Actions don't trigger from forks but I can get it most of the way there.

**Comment by shimwell** [comment]
Appears to be working over on my fork (where the GH actions run) https://github.com/Shimwell/sphinxcadquery/actions/runs/829495457

**Comment by shimwell** [comment]
Wondering if you have the  "Squash and merge" option on this on a pull request

https://i.stack.imgur.com/8t1OJ.png

File: .github/workflows/CI.yml
Comment (Peque) [inline .github/workflows/CI.yml]: I am not familiar with GitHub Actions (I have only used Travis or Gitlab CI).

With this configuration, when will the pipelines be triggered?

I would like them to trigger on any PR.

File: .github/workflows/CI.yml
Comment (Peque) [inline .github/workflows/CI.yml]: Not critical, but could you add a newline at the end of the file?

File: .github/workflows/CI.yml
Comment (Peque) [inline .github/workflows/CI.yml:21]: I think it would make sense to just have one single step where you install all dependencies (CadQuery, pytest), then sphinxcadquery and then you run `pytest`. Is that fine with you too?

File: tests/test_sphinx_build.py
Comment (Peque) [inline tests/test_sphinx_build.py]: Can you get rid of the `class` definition?

**Review by Peque — CHANGES_REQUESTED** [review]

**Comment by Peque** [comment]
@Shimwell Yes, I do have the option to squash the commits myself when merging PRs.

When I do that, I become the committer, while you keep the authorship of the commit. I don't mind much, but I would rather leave your name as the committer too, if you are willing to do the clean-up-before-merge yourself. :stuck_out_tongue_winking_eye:

File: .github/workflows/CI.yml
Comment (shimwell) [reply .github/workflows/CI.yml]: > @Shimwell Yes, I do have the option to squash the commits myself when merging PRs.
> 
> When I do that, I become the committer, while you keep the authorship of the commit. I don't mind much, but I would rather leave your name as the committer too, if you are willing to do the clean-up-before-merge yourself. 😜

Good to know. Ok I shall try to squash or failing that I shall make a clean PR.

**Comment by shimwell** [comment]
Yep I can make these changes, thanks for the suggestions

**Comment by Peque** [comment]
Closing in favor of https://github.com/Peque/sphinxcadquery/pull/27.

**Pull request closed without merging**
